src/lib/protocol/README.md

```diff
@@ -741,14 +738,9 @@ pushed headers).
 — Method **datagram:payload** *pointer*, *length*
 
 Combined payload accessor and setter method. Returns a pointer to the
-datagram payload, its byte size and the `packet.iovec` index of the next
-buffer if the payload is incomplete or `nil`.
-
-The payload is incomplete if the underlying packet consists of more than
-one buffer. The caller has to access the remainder of the payload,
-e.g. any additional packet buffers, by his own.
+datagram payload and its byte size.
 
 If *pointer* and *length* are supplied then *length* bytes starting from
-*pointer* are appended to the datagram's payload. If there is not enough
-free space in the buffer to append *length* bytes or if the payload is
-incomplete an error will be raised.
+*pointer* are appended to the datagram's payload. The caller has to
```

Comment:
The overflow is checked for in `packet.append` (used by `datagram:payload`).

> The caller has to ensure that the underlying packet has enough free space to accommodate for *length* additional bytes.

I meant this to be read as "can raise an error on overflow". Should I reword to make that meaning more implicit?